src/interp.rs

```diff
@@ -55,97 +59,160 @@ pub struct TypedVal<'tcx> {
+                // XXX: NotImplemented
+            }
+            _ => unimplemented!()
+        }
+    }
 
-fn eval_statement<'a, 'tcx>(block: BasicBlock,
-                  statement: &Statement<'tcx>,
-                  vm: &mut Machine<'a, 'tcx>) {
-    match statement.kind {
-        StatementKind::Assign(ref place, ref rvalue) => {
-            eval_assign(block, place, rvalue, vm)
+    /// The RHS of an assignment is first evaluated to a primitive. This will be
+    /// either a pointer into memory or a primitive. It is then mapped to the
+    /// required place in memory.
+    fn eval_assign(&mut self,
+                   _block: BasicBlock,
+                   place: &Place<'tcx>,
+                   rvalue: &Rvalue<'tcx>) {
+        let dest = self.eval_place(place);
+        let ref dest_ty = place.ty(self.cur_frame().mir, self.tcx).to_ty(self.tcx);
+        match *rvalue {
+            Rvalue::Use(ref operand) => {
+                let val = self.eval_operand(operand).val;
+                let tyval = TypedVal {
+                    ty: dest_ty,
+                    val: val
+                };
+                self.store(tyval, dest);
+            },
+            Rvalue::Aggregate(ref kind, ref ops) => {
+                match **kind {
+                    AggregateKind::Tuple => {
+                        // XXX: Tuple layout needs some proper thought, and
+                        // should be upcoming in the next PR, so we handle the
+                        // empty tuple explicitly as this is the bottom type in
+                        // Rust.
+                        if !ops.len() == 0 {
+                            unimplemented!()
+                        }
+                    },
+                    _ => unimplemented!()
+                }
+            },
+            _ => unimplemented!()
         }
-        _ => unimplemented!()
     }
-}
 
-/// The RHS of an assignment is first evaluated to a primitive. This will be
-/// either a pointer into memory or a primitive. It is then mapped to the
-/// required place in memory.
-fn eval_assign<'a, 'tcx>(_block: BasicBlock,
-                         place: &Place<'tcx>,
-                         rvalue: &Rvalue<'tcx>,
-                         vm: &mut Machine<'a, 'tcx>) {
-    let dest = eval_place(place);
-    let ref dest_ty = place.ty(vm.cur_frame().mir, vm.tcx).to_ty(vm.tcx);
-    match *rvalue {
-        Rvalue::Use(ref operand) => {
-            let val = eval_operand(operand, vm).val;
-            let tyval = TypedVal {
-                ty: dest_ty,
-                val: val
-            };
-            vm.store(tyval, dest);
+    /// Each `BasicBlock` is succeeded by a `Terminator`, which encapsulates control
+    /// flow. A `Terminator` determines the next `BasicBlock` that should be jumped to
+    /// during execution.
+    fn eval_terminator(&mut self,
+                       terminator: &Terminator<'tcx>) {
+        let Terminator {
```

Comment:
Isn't this equivalent to `left kind = &terminator.kind`?

Reply:
Yes, I think you've picked up on this before. So I think this might have been a regression from a botched rebase. Fixed